### PR TITLE
Introduce showInDashboard instead of showIn

### LINKED ISSUE #7642: Unresolvable showIn deprecation case
#### Sonata packages

<details><summary>show</summary>
<p>

```
$ composer show --latest 'sonata-project/*'
sonata-project/admin-bundle              3.105.0 4.5.0  The missing Symfony Admin Generator
sonata-project/block-bundle              3.23.0  4.9.0  Symfony SonataBlockBundle
sonata-project/cache                     2.1.1   2.2.0  Cache library
sonata-project/core-bundle               3.20.0  3.20.0 Symfony SonataCoreBundle (abandoned)
Package sonata-project/core-bundle is abandoned, you should avoid using it. No replacement was suggested.
sonata-project/doctrine-extensions       1.12.0  1.16.0 Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.35.0  4.2.1  Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/entity-audit-bundle       1.5.0   1.6.1  Audit for Doctrine Entities
sonata-project/exporter                  2.6.2   2.9.1  Lightweight Exporter library
sonata-project/form-extensions           0.1.2   1.12.4 Symfony form extensions
sonata-project/intl-bundle               2.7.0   2.11.2 Symfony SonataIntlBundle
sonata-project/twig-extensions           0.1.1   1.9.1  Sonata twig extensions
```

</p>
</details>

#### Symfony packages

<details><summary>show</summary>
<p>

```
$ composer show --latest 'symfony/*'
symfony/asset                      v5.2.4  v5.4.0  Manages URL generation and versioning of web assets such as CSS stylesheets, JavaScript files and ima...
symfony/browser-kit                v5.2.0  v5.4.0  Symfony BrowserKit Component
symfony/cache                      v5.2.6  v5.4.0  Provides an extended PSR-6, PSR-16 (and tags) implementation
symfony/cache-contracts            v2.4.0  v2.5.0  Generic abstractions related to caching
symfony/config                     v4.4.30 v5.4.0  Helps you find, load, combine, autofill and validate configuration values of any kind
symfony/console                    v4.4.30 v5.4.0  Eases the creation of beautiful and testable command line interfaces
symfony/contracts                  v1.1.0  v2.5.0  A set of abstractions extracted out of the Symfony components
symfony/css-selector               v5.0.8  v5.4.0  Symfony CssSelector Component
symfony/debug                      v4.4.20 v4.4.31 Provides tools to ease debugging PHP code
symfony/debug-bundle               v4.4.8  v5.4.0  Symfony DebugBundle
symfony/dependency-injection       v4.4.21 v5.4.0  Allows you to standardize and centralize the way objects are constructed in your application
symfony/deprecation-contracts      v2.4.0  v2.5.0  A generic function and convention to trigger deprecation notices
symfony/doctrine-bridge            v4.4.31 v5.4.0  Provides integration for Doctrine with various Symfony components
symfony/dom-crawler                v5.2.0  v5.4.0  Symfony DomCrawler Component
symfony/dotenv                     v4.4.8  v5.4.0  Registers environment variables from a .env file
symfony/error-handler              v4.4.21 v5.4.0  Provides tools to manage errors and ease debugging PHP code
symfony/event-dispatcher           v4.4.20 v5.4.0  Provides tools that allow your application components to communicate with each other by dispatching e...
symfony/event-dispatcher-contracts v1.1.9  v2.5.0  Generic abstractions related to dispatching event
symfony/expression-language        v4.4.20 v5.4.0  Provides an engine that can compile and evaluate expressions
symfony/filesystem                 v4.4.27 v5.4.0  Provides basic utilities for the filesystem
symfony/finder                     v5.2.4  v5.4.0  Finds files and directories via an intuitive fluent interface
symfony/flex                       v1.11.0 v1.17.6 Composer plugin for Symfony
symfony/form                       v4.4.21 v5.4.0  Allows to easily create, process and reuse HTML forms
symfony/framework-bundle           v4.4.34 v5.4.0  Provides a tight integration between Symfony components and the Symfony full-stack framework
symfony/http-client                v5.3.2  v5.4.0  Provides powerful methods to fetch HTTP resources synchronously or asynchronously
symfony/http-client-contracts      v2.4.0  v2.5.0  Generic abstractions related to HTTP clients
symfony/http-foundation            v4.4.20 v5.4.0  Defines an object-oriented layer for the HTTP specification
symfony/http-kernel                v4.4.21 v5.4.0  Provides a structured process for converting a Request into a Response
symfony/inflector                  v5.2.6  v5.4.0  Converts words between their singular and plural forms (English only)
Package symfony/inflector is abandoned, you should avoid using it. Use use `EnglishInflector` from the String component instead instead.
symfony/intl                       v4.4.20 v5.4.0  Provides a PHP replacement layer for the C intl extension that includes additional data from the ICU ...
symfony/lock                       v4.4.8  v5.4.0  Symfony Lock Component
symfony/messenger                  v4.4.19 v5.4.0  Helps applications send and receive messages to/from other applications or via message queues
symfony/mime                       v5.2.6  v5.4.0  Allows manipulating MIME messages
symfony/monolog-bridge             v5.0.8  v5.4.0  Symfony Monolog Bridge
symfony/monolog-bundle             v3.5.0  v3.7.1  Symfony MonologBundle
symfony/options-resolver           v4.4.20 v5.4.0  Provides an improved replacement for the array_replace PHP function
symfony/orm-pack                   v2.1.0  v2.1.0  A pack for the Doctrine ORM
symfony/phpunit-bridge             v4.4.8  v6.0.0  Symfony PHPUnit Bridge
symfony/polyfill-ctype             v1.23.0 v1.23.0 Symfony polyfill for ctype functions
symfony/polyfill-iconv             v1.15.0 v1.23.0 Symfony polyfill for the Iconv extension
symfony/polyfill-intl-grapheme     v1.22.1 v1.23.1 Symfony polyfill for intl's grapheme_* functions
symfony/polyfill-intl-icu          v1.22.1 v1.23.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-intl-idn          v1.22.1 v1.23.0 Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-intl-normalizer   v1.22.1 v1.23.0 Symfony polyfill for intl's Normalizer class and related functions
symfony/polyfill-mbstring          v1.23.1 v1.23.1 Symfony polyfill for the Mbstring extension
symfony/polyfill-php72             v1.23.0 v1.23.0 Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/polyfill-php73             v1.23.0 v1.23.0 Symfony polyfill backporting some PHP 7.3+ features to lower PHP versions
symfony/polyfill-php80             v1.23.1 v1.23.1 Symfony polyfill backporting some PHP 8.0+ features to lower PHP versions
symfony/polyfill-php81             v1.23.0 v1.23.0 Symfony polyfill backporting some PHP 8.1+ features to lower PHP versions
symfony/polyfill-uuid              v1.23.0 v1.23.0 Symfony polyfill for uuid functions
symfony/process                    v4.4.8  v5.4.0  Symfony Process Component
symfony/property-access            v4.4.20 v5.4.0  Provides functions to read and write from/to an object or array using a simple string notation
symfony/property-info              v5.2.4  v5.4.0  Extracts information about PHP class' properties using metadata of popular sources
symfony/proxy-manager-bridge       v4.4.20 v5.4.0  Provides integration for ProxyManager with various Symfony components
symfony/psr-http-message-bridge    v1.3.0  v2.1.2  PSR HTTP message bridge
symfony/routing                    v4.4.20 v5.4.0  Maps an HTTP request to a set of configuration variables
symfony/security                   v4.4.26 v4.4.34 Provides a complete security system for your web application
symfony/security-acl               v3.1.1  v3.2.0  Symfony Security Component - ACL (Access Control List)
symfony/security-bundle            v4.4.21 v5.4.0  Provides a tight integration of the Security component into the Symfony full-stack framework
symfony/serializer                 v4.4.20 v5.4.0  Handles serializing and deserializing data structures, including object graphs, into array structures...
symfony/serializer-pack            v1.0.4  v1.1.0  A pack for the Symfony serializer
symfony/service-contracts          v2.4.0  v2.5.0  Generic abstractions related to writing services
symfony/stopwatch                  v5.2.4  v5.4.0  Provides a way to profile code
symfony/string                     v5.2.6  v5.4.0  Provides an object-oriented API to strings and deals with bytes, UTF-8 code points and grapheme clust...
symfony/swiftmailer-bundle         v3.4.0  v3.5.3  Symfony SwiftmailerBundle
symfony/templating                 v4.4.20 v5.4.0  Provides all the tools needed to build any kind of template system
symfony/translation                v4.4.21 v5.4.0  Provides tools to internationalize your application
symfony/translation-contracts      v2.4.0  v2.5.0  Generic abstractions related to translation
symfony/twig-bridge                v4.4.21 v5.4.0  Provides integration for Twig with various Symfony components
symfony/twig-bundle                v4.4.20 v5.4.0  Provides a tight integration of Twig into the Symfony full-stack framework
symfony/validator                  v4.4.21 v5.4.0  Provides tools to validate values
symfony/var-dumper                 v4.4.21 v5.4.0  Provides mechanisms for walking through any arbitrary PHP variable
symfony/var-exporter               v5.2.4  v5.4.0  Allows exporting any serializable PHP data structure to plain PHP code
symfony/web-link                   v4.4.21 v5.4.0  Manages links between resources
symfony/web-profiler-bundle        v4.4.8  v5.4.0  Symfony WebProfilerBundle
symfony/webpack-encore-bundle      v1.7.3  v1.13.2 Integration with your Symfony app & Webpack Encore!
symfony/yaml                       v4.4.21 v5.4.0  Loads and dumps YAML files
```

</p>
</details>

#### PHP version

```
$ php -v
PHP 7.4.13 (cli) (built: Dec  1 2020 04:25:48) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.13, Copyright (c), by Zend Technologies
```

## Subject

The `Abstract::showIn` was considered as final since https://github.com/sonata-project/SonataAdminBundle/pull/7266.

According to my code search (nothing found on the upgrade), we have to use `getPermissionsShow` instead.

However, this does not fit cases where the `showIn` result does not rely on permission but on the context.

## Steps to reproduce

I refactored this logic:

```php
public function showIn($context)
{
    if ($context === AbstractAdmin::CONTEXT_DASHBOARD) {
        return false;
    }

    return parent::showIn($context);
}
```

Onto: 

```php
public function getPermissionsShow($context)
{
    if ($context === AbstractAdmin::CONTEXT_DASHBOARD) {
        return ['ROLE_NO_ACCESS'];
    }

    return parent::getPermissionsShow($context);
}
```

## Expected results

As it works with the old logic, the related admin should not appear on the dashboard context (e.g. on the create navbar super button).

## Actual results

The new logic as new effect and I see the reference of the target admin on the dashboard context.

Also. returning roles looks more hacky than the original way as the show behavior does not relate to security here.

I may understand the original usage was wrong (and old). However, it is possible to do that and breaking for v4.0. An upgrade guide for that kind of cases would be awesome.

Best regards

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

Closes #7642.

## Changelog

```markdown
### Added
- `AdminInterface::showInDashboard()` method.

### Deprecated
- `AdminInterface::showIn()` method.
```

### CODE REVIEW

**Comment by soullivaneuh** [comment]
I am unsure about the final naming, however this looks to fix the issue.

However, be aware that the issue is reported on the 3.x release. Can this be easily back ported?

**Comment by VincentLanglet** [comment]
> I am unsure about the final naming

I'm open to suggestion.
The call `showIn` was made on the `Context_Dashboard` constant so `showInDashboard` seems natural to me.
But I didn't look where it's use so maybe another name is better.

> However, be aware that the issue is reported on the 3.x release. Can this be easily back ported?

In 3.x, it's "only" a deprecation, which could be ignored.
Not sure it worth back porting this (and dealing with conflict).
When bumping to 3.x to 4.x, you'll just have to rename `showIn` to `showInDashboard`.

**Review by jordisala1991 — DISMISSED** [review]

File: src/Admin/AbstractAdmin.php [hunk 1]
```diff
@@ -1584,9 +1587,29 @@ public function getObjectIdentifier()
...
+     *
+     * @deprecated since sonata-project/admin-bundle version 4.x use showInDashboard instead
+     */
     final public function showIn(string $context): bool
     {
-        return $this->isGranted($this->getPermissionsShow($context));
+        if ('sonata_deprecation_mute' !== (\func_get_args()[1] ?? null)) {
+            @trigger_error(sprintf(
+                'The "%s()" method is deprecated since sonata-project/admin-bundle version 4.x and will be'
+                .' removed in 5.0 version. Use showInDashboard() instead',
```
Comment (phansys) [inline src/Admin/AbstractAdmin.php]:
```suggestion
                .' removed in 5.0 version. Use showInDashboard() instead.',
```

Comment (VincentLanglet) [reply src/Admin/AbstractAdmin.php]: Done

**Review by jordisala1991 — APPROVED** [review]

**Review by core23 — APPROVED** [review]

**Pull request merged**